libretro-rs/src/lib.rs

```diff
@@ -197,14 +198,17 @@ impl RetroEnvironment {
   }
 
   /// Queries a string slice from the environment. A null pointer is interpreted as `None`.
-  pub fn get_str<'a>(&'a self, key: u32) -> Option<&'a str> {
+  pub fn get_str(&self, key: u32) -> Option<&str> {
     unsafe {
       let s = self.get(key)?;
       CStr::from_ptr(s).to_str().ok()
     }
   }
 
   /// Queries a struct from the environment. A null pointer is interpreted as `None`.
+  ///
+  /// # Safety
+  /// The environment callback must be used in accordance with the libretro specification.
```

Comment:
Unless I'm mistaken, I think doc comments typically include a `Panics` section detailing the circumstances where this method will panic. Whereas `Safety` comments are used whenever using an `unsafe` block inside a safe method, explaining why it isolates the `unsafe`ty and is therefor safe.

For example:

```rust
/// # Panics
/// This method panics if the argument for `x` is zero.
fn panics_if_zero(x: i32) {
  // ..
}

fn safe_with_unsafe_internals() {
  // safety: this is actually safe overall, because ...
  unsafe {
    // ..
  }
}
```

Reply:
Clippy expects any unsafe function to have a Safety section in the documentation explaining what requirements have to be met when using it. That said, misusing any of these methods would cause undefined behavior rather than a panic.

Reply:
/sigh

And thus, clippy will continue to be ~hated~ loved by rust developers..